x/gov/legacy/v043/json.go

```diff
@@ -15,18 +24,37 @@ func migrateJSONWeightedVotes(oldVotes v040gov.Votes) types.Votes {
 	return newVotes
 }
 
+func migrateJSONProposals(oldProposals v040gov.Proposals) types.Proposals {
+	newProps := make(types.Proposals, len(oldProposals))
+	for i, oldProp := range oldProposals {
+		newProps[i] = types.Proposal{
+			ProposalId:       oldProp.ProposalId,
+			Content:          oldProp.Content,
```

Comment:
I think you need to migrate the content as well:
```
Error: error validating genesis file /home/bartleby/.gm/network2/config/genesis.json: failed to unmarshal gov genesis state: unable to resolve type URL /cosmos.upgrade.v1beta1.SoftwareUpgradeProposal
```

This is from testing genesis export/import and migration using commit `b2456e3a2a29`

I think you just need to check if `oldProp.Content.Type == /cosmos.upgrade.v1beta1.SoftwareUpgradeProposal`. Do the store migrations fix this as well? I ran a x/upgrade without this pr, but querying proposals doesn't work now (since the proposal wasn't migrated ofc):
```
 gaiad q gov proposals --node http://localhost:27000
Error: rpc error: code = Unknown desc = rpc error: code = Internal desc = rpc error: code = Internal desc = no concrete type registered for type URL /cosmos.upgrade.v1beta1.SoftwareUpgradeProposal against interface *types.Content: unknown request
```

Reply:
Maybe some sort of registry could work. I register my type url and a function which performs a migration and returns a new Any. So in this case I'd register a function to migrate `/cosmos.upgrade.v1beta1.SoftwareUpgradeProposal` to `/cosmos.upgrade.v1.SoftwareUpgradeProposal`, then the migration code could just call the registry to perform the migration, if a migration func is registered

Obviously out of scope of this pr, but seems likely the usage of `Any` in migrations needs some sort of handling/design

Reply:
Hmm, good point, I'll add logic to migrate `Any`s too from `v1beta1` to `v1`